californium-core/src/main/java/org/eclipse/californium/core/network/Matcher.java

```diff
@@ -182,11 +186,23 @@ public void sendRequest(Exchange exchange,final  Request request) {
 			}
 		}
 
-		if (!request.getOptions().hasBlock2() && request.getOptions().hasObserve() && request.getOptions().getObserve() == 0) {
+		// for observe request
+		if ( request.getOptions().hasObserve() && request.getOptions().getObserve() == 0 && !request.getOptions().hasBlock2()) {
```

Comment:
Why do we check for the absence of the Block2 option? My understanding of the [Blockwise draft 19](https://tools.ietf.org/html/draft-ietf-core-block-19#section-2.4) is that an initial GET request MAY also include this option.